data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/model/ItemInfo.java

```diff
@@ -0,0 +1,49 @@
+package org.opensearch.dataprepper.plugins.source.saas_crawler.model;
+
+
+import lombok.Getter;
+import lombok.NonNull;
+
+import java.util.Map;
+
+
+@Getter
+public abstract class ItemInfo {
+
+    /**
+     * Use this field to store primary item of a repository. Primary item of a repository is something
+     * which can be fetched/queried/obtained from repository just using its item ID.
+     */
+    @NonNull
+    String itemId;
+
+    /**
+     * Use this field to store items metadata. Item metadata can be any information other than item
+     * contents itself which can be used to apply regex filtering, change data capture etc. general
+     * assumption here is that fetching metadata should be faster than fetching entire Item
+     */
+    Map<String, String> metadata;
```

Comment:
Probably better to make it `Map<String, Object>`, it is probably unrealistic to expect all of metadata values to be Strings.

Reply:
Considering each source may have different needs, agree to make it more generic. Converted the map as suggested.